app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java

```diff
@@ -145,9 +145,9 @@ public void onError(Throwable e) {
                     public void onNext(Page<Client> clientPage) {
                         getMvpView().hideProgress();
                         if (clientPage.getPageItems().size() != 0) {
-                            long clientId = clientPage.getPageItems().get(0).getId();
-                            getMvpView().showClient(clientId);
-                            preferencesHelper.setClientId(clientId);
+                            Client clientCurrent = clientPage.getPageItems().get(0);
+                            getMvpView().showClient(clientCurrent);
```

Comment:
You do not need to pass the client id any where. check the DataManager class. As you are logging into the app. client id is passing into the DataManager So make you retro call in DataManager using the DataManager's client id.